services/wireguard/service/service.go

```diff
@@ -110,7 +114,49 @@ func (manager *Manager) ProvideConfig(publicKey json.RawMessage) (session.Servic
 		}
 	}
 
-	return config, destroy, nil
+	return config, manager.once(destroy), nil
+}
+
+// TODO this is a temporary solution to cleanup oldest used wireguard resources.
+// TODO it should be removed once payment bases session cleanup implemented.
+func (manager *Manager) once(f func()) func() {
+	manager.mu.Lock()
+	defer manager.mu.Unlock()
+
+	once := sync.Once{}
+	cleanOnce := func() {
+		once.Do(f)
+	}
+	manager.list = append(manager.list, &cleanOnce)
+
+	return func() {
+		cleanOnce()
+
+		manager.mu.Lock()
+		defer manager.mu.Unlock()
+
+		for i := range manager.list {
+			if manager.list[i] == &cleanOnce {
+				manager.list = append(manager.list[0:i], manager.list[i+1:]...)
+				return
+			}
+		}
+	}
+}
+
+// TODO this is a temporary solution to cleanup oldest used wireguard resources.
+// TODO it should be removed once payment bases session cleanup implemented.
+func (manager *Manager) cleanOldEndpoints() {
+	manager.mu.Lock()
+	defer manager.mu.Unlock()
+
+	if len(manager.list) >= resources.MaxResources-1 {
+		log.Warn(logPrefix, "We have reached a maximum number of interfaces. Cleaning up oldest one.")
+
+		f := *manager.list[0]
```

Comment:
It will be oldest endpoint, but not oldest active endpoint? I suppose it might be enough for a temp solution.